tango/step.py

```diff
@@ -58,6 +60,40 @@ def get_args(tp):  # type: ignore
 _random_for_step_names = random.Random()
 
 
+@dataclass
+class StepResources(FromParams):
+    """
+    TaskResources describe minimum external hardware requirements which must be available for a
+    step to run.
+    """
+
+    cpu_count: Optional[float] = None
+    """
+    Minimum number of logical CPU cores. It may be fractional.
+
+    Examples: ``4``, ``0.5``.
```

Comment:
What do we mean by fractional cpu cores?

Reply:
Beaker allows this, and I think it's just passed on to the `docker run ...` command as-is.

https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler